### PR TITLE
ftxui: 3.0.0 -> 4.1.0

### PR DESCRIPTION
**Pull request by hzeller**
###### Description of changes

Update of ftxui to version already released back in February.

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [x] aarch64-linux
  - [x] x86_64-darwin
  - [x] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by hzeller** [comment]
In the meantime, I've upstreamed little changes to make tests work in aarch64 and also add a pkg-config file; this is now upstream release 4.1.0. Since this PR is still open, updated not only to 4.0.0, but 4.1.0 already :)

**Review by tobim — APPROVED** [review]

**Comment by mavam** [comment]
Looking forward to get this merged!

**Comment by hzeller** [comment]
Do we need more approvals for this to come to the attention of someone who can merge this ?

**Comment by hzeller** [comment]
Adding @marsam  who merged the last ftxui PR #181000

File: pkgs/development/libraries/ftxui/default.nix [hunk 1]
```diff
@@ -1,51 +1,45 @@
...
-      sha256 = "sha256-0040/gJcCXzL92FQLhZ2dNMJhNqXXD+UHFv4Koc07K0=";
-    })
-  ];
-
   nativeBuildInputs = [
     cmake
     doxygen
+    gbenchmark
     graphviz
+    gtest
```
Comment (marsam) [inline pkgs/development/libraries/ftxui/default.nix:27]: I think `gtest` and `gbenchmark` should be in nativeCheckInputs
```nix
nativeCheckInputs = [
  gbenchmark
  gtest
];
```

File: pkgs/development/libraries/ftxui/default.nix [hunk 2]
```diff
@@ -1,51 +1,45 @@
...
   ];
 
   cmakeFlags = [
     "-DFTXUI_BUILD_EXAMPLES=OFF"
+    "-DFTXUI_BUILD_TESTS=On"
```
Comment (marsam) [inline pkgs/development/libraries/ftxui/default.nix:32]: nitpick
```suggestion
    "-DFTXUI_BUILD_TESTS=ON"
```

File: pkgs/development/libraries/ftxui/default.nix (hunk 1 above)
Comment (hzeller) [reply pkgs/development/libraries/ftxui/default.nix:30]: done.

File: pkgs/development/libraries/ftxui/default.nix (hunk 2 above)
Comment (hzeller) [reply pkgs/development/libraries/ftxui/default.nix]: Done.

**Pull request merged**